### PR TITLE
`lima`: 0.22.1 -> 1.0.1 (also fixes `vz`)

### PR DESCRIPTION
**Pull request by booxter**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [X] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/virtualization/lima/default.nix [hunk 1]
```diff
@@ -7,29 +7,34 @@
...
   };
 
-  vendorHash = "sha256-P0Qnfu/cqLveAwz9jf/wTXxkoh0jvazlE5C/PcUrWsA=";
+  passthru.updateScript = nix-update-script { };
 
-  nativeBuildInputs = [ makeWrapper installShellFiles ]
+  vendorHash = "sha256-nNSBwvhKSWs6to37+RLziYQqVOYfvjYib3fRRALACho=";
+
+  nativeBuildInputs = [ makeWrapper installShellFiles patchelf ]
     ++ lib.optionals stdenv.hostPlatform.isDarwin [ xcbuild.xcrun sigtool ];
```
Comment (emilazy) [inline pkgs/applications/virtualization/lima/default.nix:31]: `xcbuild.xcrun` is no longer required. What’s `patchelf` for?

File: pkgs/applications/virtualization/lima/default.nix [hunk 2]
```diff
@@ -7,29 +7,34 @@
...
   src = fetchFromGitHub {
     owner = "lima-vm";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-ZX2FSZz9q56zWPSHPvXUOf2lzBupjgdTXgWpH1SBJY8=";
+    sha256 = "sha256-XYB8Nxbs76xmiiZ7IYfgn+UgUr6CLOalQrl6Ibo+DRc=";
   };
 
-  vendorHash = "sha256-P0Qnfu/cqLveAwz9jf/wTXxkoh0jvazlE5C/PcUrWsA=";
+  passthru.updateScript = nix-update-script { };
```
Comment (emilazy) [inline pkgs/applications/virtualization/lima/default.nix:26]: Nit: This conventionally goes down before `meta`.

File: pkgs/applications/virtualization/lima/default.nix [hunk 3]
```diff
@@ -7,29 +7,34 @@
...
     ++ lib.optionals stdenv.hostPlatform.isDarwin [ xcbuild.xcrun sigtool ];
 
-  # clean fails with read only vendor dir
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin [ apple-sdk_13 ];
+
   postPatch = ''
     substituteInPlace Makefile \
-      --replace 'binaries: clean' 'binaries:' \
-      --replace 'codesign --entitlements vz.entitlements -s -' 'codesign --force --entitlements vz.entitlements -s -'
+      --replace-fail 'codesign -f -v --entitlements vz.entitlements -s -' 'codesign -f --entitlements vz.entitlements -s -'
```
Comment (emilazy) [inline pkgs/applications/virtualization/lima/default.nix:37]: Does this work properly on `aarch64-darwin`? I thought you could only do this ad‐hoc entitlements stuff on the same machine the resulting binary runs on, or something. (Could be totally wrong.)

**Review by emilazy — COMMENTED** [review]
Looks good! The [Go `Virtualization.framework` library](https://github.com/Code-Hex/vz) they use has availability checks up to macOS 15, so I think `apple-sdk_15` is the best choice here for full functionality.

File: pkgs/applications/virtualization/lima/default.nix (hunk 3 above)
Comment (booxter) [reply pkgs/applications/virtualization/lima/default.nix:37]: I am not sure what "works" means here. @emilazy could you please elaborate? I can use the tool to start VMs, but build the derivation on this same machine where I run it, so perhaps I miss something. How do I check if it works? (Note that this `--replace` was present before the bump.)

Comment (emilazy) [reply pkgs/applications/virtualization/lima/default.nix:33]: It’s probably fine :) I think I’m probably just thinking of the more exclusive entitlements.

**Review by emilazy — APPROVED** [review]
Thanks!

**Comment by kachick** [comment]
## `nixpkgs-review` result
 
Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).
 
Command: `nixpkgs-review pr 355729`
 
---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>colima</li>
    <li>lima</li>
  </ul>
</details>

### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>ruffle</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>lima</li>
  </ul>
</details>

### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>lima</li>
  </ul>
</details>

**Comment by emilazy** [comment]
Hmm, this should probably set `meta.platforms`…

**Comment by booxter** [comment]
@emilazy Why? I *think* lima can be used on linux too (even if it was primarily targeted for MacOS users). What do I miss?

**Comment by emilazy** [comment]
Huh, apparently so. However going by the log above it does not actually build in practice, so either that should be fixed or `meta.badPlatforms` should probably be set.

**Pull request merged**